scrapy/exporters.py

```diff
@@ -86,10 +88,10 @@ def __init__(self, file, **kwargs):
 
     def export_item(self, item):
         itemdict = dict(self._get_serialized_fields(item))
-        self.file.write(self.encoder.encode(itemdict) + '\n')
+        self.file.write(to_bytes(self.encoder.encode(itemdict) + '\n'))
 
 
-class JsonItemExporter(JsonLinesItemExporter):
+class JsonItemExporter(BaseItemExporter):
```

Comment:
`JsonItemExporter` and `JsonLinesItemExporter` still share common stuff.
what's the reason for having them both subclass `BaseItemExporter`?